istioctl/cmd/proxystatus.go

```diff
@@ -89,3 +93,73 @@ func newKubeClientWithRevision(kubeconfig, configContext string, revision string
 func newKubeClient(kubeconfig, configContext string) (kube.ExtendedClient, error) {
 	return newKubeClientWithRevision(kubeconfig, configContext, "")
 }
+
+func xdsStatusCommand() *cobra.Command {
+	var opts clioptions.ControlPlaneOptions
+	var centralOpts clioptions.CentralControlPlaneOptions
+
+	statusCmd := &cobra.Command{
+		Use:   "proxy-status [<pod-name[.namespace]>]",
+		Short: "Retrieves the synchronization status of each Envoy in the mesh",
+		Long: `
+Retrieves last sent and last acknowledged xDS sync from Istiod to each Envoy in the mesh
+
+`,
+		Example: `# Retrieve sync status for all Envoys in a mesh
+	istioctl proxy-status
+
+# Retrieve sync diff for a single Envoy and Istiod
+	istioctl proxy-status istio-egressgateway-59585c5b9c-ndc59.istio-system
+`,
+		Aliases: []string{"ps"},
+		RunE: func(c *cobra.Command, args []string) error {
+			kubeClient, err := kubeClientWithRevision(kubeconfig, configContext, opts.Revision)
+			if err != nil {
+				return err
+			}
+
+			if len(args) > 0 {
+				podName, ns := handlers.InferPodInfo(args[0], handlers.HandleNamespace(namespace, defaultNamespace))
+				path := "config_dump"
+				envoyDump, err := kubeClient.EnvoyDo(context.TODO(), podName, ns, "GET", path, nil)
+				if err != nil {
+					return err
+				}
+
+				xdsRequest := xdsapi.DiscoveryRequest{
+					Node: &envoy_corev3.Node{
+						Id: "debug~0.0.0.0~istioctl~cluster.local",
+					},
+					TypeUrl: fmt.Sprintf("%s?proxyID=%s.%s", pilotxds.TypeDebugConfigDump, podName, ns),
+				}
+				xdsResponses, err := multixds.AllRequestAndProcessXds(&xdsRequest, &centralOpts, istioNamespace, kubeClient)
```

Comment:
It might be optimal to short circuit this function as soon as it has found a non-null response.  This makes it more efficient, and also prevents the case where two istiod's return config for a particular Envoy (which is possible immediately after Envoy rebalances).

Reply:
Fixed.